scipy/optimize/tests/test_least_squares.py

```diff
@@ -400,6 +400,7 @@ def test_convergence_with_only_one_tolerance_enabled(self):
                                 method=self.method)
             assert_allclose(res.x, x_opt)
 
+    @pytest.mark.parallel_threads(4)  # 0.4 GiB per thread RAM usage
```

Comment:
This is probably sensible enough. Note that when I run this test locally on `main` on x86_64 Linux with Python `3.13.0` free-threaded via `spin test -t scipy/optimize/tests/test_least_squares.py -- -k "test_workers" --parallel-threads=32`, I actually see a `multiprocessing` error about too many open files, and sometimes a segfault too.

So, that seems like two reasons for this patch I suppose (the errors go away with the thread limiting as well).

Reply:
Presumably mixing `Pool()` with free-threading may cause some resource issues with sufficiently high free-threading concurrency.